File: cash.py
from cs50 import get_float

#set the money counter
count = 0

# as the situation is true continue
while True:
    change = get_float("The change is: ")
    if change > 0:
        break
    
dollarCent = round(change * 100)

#checking for 0.25 dollars
while dollarCent >= 25:
    dollarCent = dollarCent - 25
    count += 1

#checkin' for 0.10 dollars
while dollarCent >= 10:
    dollarCent = dollarCent - 10
    count += 1
    
# checkin' for 0.05 dollars 
while dollarCent >= 5:
    dollarCent = dollarCent - 5
    count += 1

#checking for 0.01 dollars
while dollarCent >= 1:
    dollarCent = dollarCent - 1
    count += 1
    
# printing the sum
print(count)

File: cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void)
{
    // Ask for input
    float change;
    do
    {
        change = get_float("The change is: ");
    }
    while (change < 0);
    
    // Round cents to the nearest penny
    
    int dollarCent = round (change * 100);
    int money = 0;
    
    // Computing the quarters
    
    while (dollarCent >= 25)
    {
        dollarCent = dollarCent - 25;
        money++;
    }
    
    // Computing the dimes
    
    while (dollarCent >= 10)
    {
        dollarCent = dollarCent - 10;
        money++;
    }
    
    // Computing the nickels
    
    while (dollarCent >= 5)
    {
        dollarCent = dollarCent - 5;
        money++;
    }
    
    // Computing the pennies 'till change is equal to 0
    
    while (dollarCent >= 1 && dollarCent > 0)
    {
        dollarCent = dollarCent - 1;
        money++;
    }
    
    // At the end, print the answer of this question "how many coins does the cashier give?"
    
    printf("%i\n", money);
}